policyengine_us/variables/gov/states/ny/tax/income/ny_supplemental_tax.py

```diff
@@ -53,29 +53,33 @@ def formula(tax_unit, period, parameters):
             applicable_amount / p.phase_in_length,
         )
 
-        recapture_base = select(
-            in_each_status,
-            [
-                p.recapture_base.single.calc(ny_taxable_income),
-                p.recapture_base.joint.calc(ny_taxable_income),
-                p.recapture_base.head_of_household.calc(ny_taxable_income),
-                p.recapture_base.widow.calc(ny_taxable_income),
-                p.recapture_base.separate.calc(ny_taxable_income),
-            ],
-        )
+        if p.in_effect:
+            recapture_base = select(
+                in_each_status,
+                [
+                    p.recapture_base.single.calc(ny_taxable_income),
+                    p.recapture_base.joint.calc(ny_taxable_income),
+                    p.recapture_base.head_of_household.calc(ny_taxable_income),
+                    p.recapture_base.widow.calc(ny_taxable_income),
+                    p.recapture_base.separate.calc(ny_taxable_income),
+                ],
+            )
 
-        incremental_benefit = select(
-            in_each_status,
-            [
-                p.incremental_benefit.single.calc(ny_taxable_income),
-                p.incremental_benefit.joint.calc(ny_taxable_income),
-                p.incremental_benefit.head_of_household.calc(
-                    ny_taxable_income
-                ),
-                p.incremental_benefit.widow.calc(ny_taxable_income),
-                p.incremental_benefit.separate.calc(ny_taxable_income),
-            ],
-        )
+            incremental_benefit = select(
+                in_each_status,
+                [
+                    p.incremental_benefit.single.calc(ny_taxable_income),
+                    p.incremental_benefit.joint.calc(ny_taxable_income),
+                    p.incremental_benefit.head_of_household.calc(
+                        ny_taxable_income
+                    ),
+                    p.incremental_benefit.widow.calc(ny_taxable_income),
+                    p.incremental_benefit.separate.calc(ny_taxable_income),
+                ],
+            )
+        else:
+            recapture_base = 0
```

Comment:
add a test for 2028 showing that the benefit is 0

Reply:
we should make this its own variable which then return 0 in 2028 ny_supplemental_tax_incremental_benefit.py

Reply:
I added two unit tests for 2028 and slightly modified the formula to align with similar functions.